pkg/nodeconfig/config.go

```diff
@@ -167,6 +168,70 @@ func (m *NodeConfig) Restore() error {
 	return extractConfig(m.fullMachinePath, data)
 }
 
+// UpdateAmazonAuth updates the machine config.json file on disk with the most
+// recent version of creds from the cloud credential
+func (m *NodeConfig) UpdateAmazonAuth(rawConfig interface{}) (bool, error) {
+	var update bool
+	c := convert.ToMapInterface(rawConfig)
+
+	machines := filepath.Join(m.fullMachinePath, "machines")
+
+	files, err := ioutil.ReadDir(machines)
+	if err != nil {
+		// There aren't any machines, nothing to update
+		if os.IsNotExist(err) {
+			return update, nil
+		}
+		return update, err
+	}
+
+	for _, file := range files {
+		if file.IsDir() {
+			configPath := filepath.Join(machines, file.Name(), "config.json")
+			b, err := ioutil.ReadFile(configPath)
+			if err != nil {
+				if os.IsNotExist(err) {
+					continue
```

Comment:
Nit: should we log the error, in debug at least?

Reply:
The error indicates we don't have any machines yet which is a valid state